database/test/db-admin-persona.test.js

```diff
@@ -80,4 +80,48 @@ describe('Admin persona view', () => {
       assert.equal(personaPermissions.length, 0)
     })
   })
+
+  describe('Test Insert personas permissions', () => {
+    let testPersonaKey
+    let validPermissionKey
+    const invalidPersonaKey = 0
+    before(async () => {
+      const { key: permissionKey } = await usherDb('permissions').select('key').first()
+      validPermissionKey = permissionKey
+      const { key: tenantkey } = await usherDb('tenants').select('key').first()
+      const [persona] = await usherDb('personas').insert({ tenantkey, sub_claim: 'personapermission@test' }).returning('key')
+      testPersonaKey = persona.key
+    })
+
+    it('Should return an array of inserted personapermissions records', async () => {
+      const personaPermissions = await adminPersonas.insertPersonaPermissions(testPersonaKey, [validPermissionKey])
+      assert.equal(personaPermissions.length, 1)
+      assert.equal(personaPermissions[0].personakey, testPersonaKey)
+      assert.equal(personaPermissions[0].permissionkey, validPermissionKey)
+    })
+
+    it('Should fail due to invalid persona key', async () => {
+      try {
+        await adminPersonas.insertPersonaPermissions(invalidPersonaKey, [validPermissionKey])
+      } catch (err) {
+        assert.equal(!!err, true)
+      }
+    })
+
+    it('Should fail due to duplicate permission', async () => {
```

Comment:
@k1ch do you think this should fail with an error or should we make this function idempotent and return as the end state is the persona has the permission assigned?

Reply:
So let's consider 2 scenarios: 
1. All permissions in the list are already assigned to persona: should we return 200?
2. Some of permissions are not assigned: should we assign those and return 201 and ignore the duplicates?

Reply:
@k1ch  I am ok considering this for a separate PR, but to answer your question I would consider both of those use cases to have the same end result. 

If a caller is making a request to assign X permissions to a user, if the actual work required to assign those permissions is less than X, the final result though is still all X permissions are assigned. 

This is assuming no other logical error like an invalid persona or invalid permission Id, those for sure should return errors 400 bad request errors.

Reply:
Sure. To clarify, currently an invalid `personakey` causes `404` and invalid `permissionkeys` causes `400` 
```json
{
    "code": 400,
    "message": "Referenced resource is invalid!"
}
```
To implement your proposal we need to first `SELECT key FROM permissions WHERE key IN (Permission list provided in payload);` and return `400` if some of permission keys are invalid.